code/datums/components/slippery.dm

```diff
@@ -0,0 +1,54 @@
+/**
+	This is a component that can be applied to any movable atom (mob or obj).
+
+	While the atom has this component, any carbon mob that walks over it will slip.
+	Duration, tiles moved, and so on, depend on what variables are passed in when the component is added.
+*/
+/datum/component/slippery
+	/// Text that gets displayed in the slip proc, i.e. "user slips on [description]"
+	var/description
+	/// The amount of stun to apply after slip.
+	var/stun
+	/// The amount of weaken to apply after slip.
+	var/weaken
+	/// The chance that walking over the parent will slip you.
+	var/slip_chance
+	/// The amount of tiles someone will be moved after slip.
+	var/slip_tiles
+	/// TRUE If this slip can be avoided by walking.
+	var/walking_is_safe
+	/// TRUE if having no slip shoes makes you immune to this slip.
+	var/noslip_is_immune
+	/// The verb that players will see when someone slips on the parent. In the form of "You [slip_verb]ped on".
+	var/slip_verb
```

Comment:
The documentation is great for the var definitions, but not for the definition.

While I would love to suggest reading up on the doucmentation guide, its not compulsory here so its no huge issue

Reply:
I can make this *not* autodoc'd if you want, as in `/* bla */`

Reply:
```
/**
	This is a component that can be applied to any movable atom (mob or obj).
	While the atom has this component, any carbon mob that walks over it will slip.
	Duration, tiles moved, and so on, depend on what variables are passed in when the component is added.
*/
```

Should be
```
/**
  * # Slip Component
  *
  * This is a component that can be applied to any movable atom (mob or obj).
  *
  * While the atom has this component, any carbon mob that walks over it will slip.
  * Duration, tiles moved, and so on, depend on what variables are passed in when the component is added.
  *
  */
```

but again, doc isnt required here so its up to you

Reply:
Eh I really hate that asterisk per line format but, If that's how its suppose to be, why not